ospf6d/ospf6_interface.c

```diff
@@ -1402,82 +1421,156 @@ DEFUN(show_ipv6_ospf6_interface_traffic,
+		for (ALL_LIST_ELEMENTS_RO(om6->ospf6, node, ospf6)) {
+			ifp = if_lookup_by_name(argv[idx_ifname]->arg,
+						ospf6->vrf_id);
+			if (ifp == NULL) {
+				vty_out(vty, "No such Interface: %s\n",
+					argv[idx_ifname]->arg);
+				return CMD_WARNING;
+			}
 
-	ifp = if_lookup_by_name(argv[idx_ifname]->arg, VRF_DEFAULT);
+			oi = ifp->info;
+			if (oi == NULL) {
+				vty_out(vty, "OSPFv3 is not enabled on %s\n",
+					argv[idx_ifname]->arg);
+				return CMD_WARNING;
+			}
+			ospf6_route_table_show(vty, idx_prefix, argc, argv,
+					       oi->route_connected);
+		}
+		return CMD_WARNING;
+	}
+	ospf6 = ospf6_lookup_by_vrf_name(vrf_name);
+	OSPF6_CMD_CHECK_RUNNING(ospf6);
+	ifp = if_lookup_by_name(argv[idx_ifname]->arg, ospf6->vrf_id);
 	if (ifp == NULL) {
 		vty_out(vty, "No such Interface: %s\n", argv[idx_ifname]->arg);
 		return CMD_WARNING;
 	}
-
 	oi = ifp->info;
 	if (oi == NULL) {
 		vty_out(vty, "OSPFv3 is not enabled on %s\n",
 			argv[idx_ifname]->arg);
 		return CMD_WARNING;
 	}
-
 	ospf6_route_table_show(vty, idx_prefix, argc, argv,
 			       oi->route_connected);
 
 	return CMD_SUCCESS;
 }
 
-DEFUN (show_ipv6_ospf6_interface_prefix,
-       show_ipv6_ospf6_interface_prefix_cmd,
-       "show ipv6 ospf6 interface prefix\
+DEFUN(show_ipv6_ospf6_interface_prefix, show_ipv6_ospf6_interface_prefix_cmd,
+      "show ipv6 ospf6 [vrf <NAME|all>] interface prefix\
           [<\
 	    detail\
 	    |<X:X::X:X|X:X::X:X/M> [<match|detail>]\
 	  >]",
-       SHOW_STR
-       IP6_STR
-       OSPF6_STR
-       INTERFACE_STR
-       "Display connected prefixes to advertise\n"
-       "Display details of the prefixes\n"
-       OSPF6_ROUTE_ADDRESS_STR
-       OSPF6_ROUTE_PREFIX_STR
-       OSPF6_ROUTE_MATCH_STR
-       "Display details of the prefixes\n")
+      SHOW_STR IP6_STR OSPF6_STR VRF_CMD_HELP_STR
+      "All VRFs\n" INTERFACE_STR
+      "Display connected prefixes to advertise\n"
+      "Display details of the prefixes\n" OSPF6_ROUTE_ADDRESS_STR
+	      OSPF6_ROUTE_PREFIX_STR OSPF6_ROUTE_MATCH_STR
+      "Display details of the prefixes\n")
 {
-	struct vrf *vrf = vrf_lookup_by_id(VRF_DEFAULT);
+	struct vrf *vrf = NULL;
 	int idx_prefix = 5;
 	struct ospf6_interface *oi;
 	struct interface *ifp;
+	struct listnode *node;
+	struct ospf6 *ospf6;
+	const char *vrf_name = VRF_DEFAULT_NAME;
+	bool all_vrf = false;
+	int idx_vrf = 0;
+
+	OSPF6_FIND_VRF_ARGS(argv, argc, idx_vrf, vrf_name, all_vrf);
+	if (idx_vrf > 0)
+		idx_prefix += 2;
+
+	if (all_vrf) {
+		for (ALL_LIST_ELEMENTS_RO(om6->ospf6, node, ospf6)) {
+			vrf = vrf_lookup_by_id(ospf6->vrf_id);
+			FOR_ALL_INTERFACES (vrf, ifp) {
+				oi = (struct ospf6_interface *)ifp->info;
+				if (oi == NULL)
+					continue;
 
+				ospf6_route_table_show(vty, idx_prefix, argc,
+						       argv,
+						       oi->route_connected);
+			}
+		}
+		return CMD_SUCCESS;
+	}
+	ospf6 = ospf6_lookup_by_vrf_name(vrf_name);
+	OSPF6_CMD_CHECK_RUNNING(ospf6);
+	vrf = vrf_lookup_by_id(ospf6->vrf_id);
```

Comment:
Same comment as before. This call for specific VRFs could also have been accommodated inside the for loop for 'all_vrf'.

Reply:
Resolved.